### ISSUE TITLE
Array of references 

### ISSUE THREAD

**Issue by Dragomir-Ivanov** [rs/rest-layer#222]
Hi @smyrman ,
I saw about #100, however the API has changed since then, how this is supposed to be configured:
```
			"applianceIds": {
				Validator: &schema.Array{
					Values: schema.Field{Validator: &schema.Reference{Path: "appliances"}},
				},
			},
```
Thanks.

**Comment by smyrman** [comment]
I believe that will just work for the validation part, but hooks won't get passed the right context as described in #192, and field embedding probably won't work; that wasn't solved by #100. Field embedding still has a hardcoroded type switch.

**Comment by Dragomir-Ivanov** [comment]
Thanks Sindre. However I need `field embedding` part. It seems that this case is the only one missing from `rest-layer. I am willing to work on this. 
Do you have anything in mind? Maybe I can try hard-code `schema.Reference in schema.Array` in `evalProjection` function?

**Comment by smyrman** [comment]
What we have done for similar cases when embedding is needed, is too implement it as a field handler parameter, e.g. expand=1 (one level), expand=2 (two levels).

For fixing it properly, some sort of interface is needed that the code doing the embedding can use.

**Comment by Dragomir-Ivanov** [comment]
It would be good if mainline `rest-layer` supports such `expand` query parameter. However often times not full expansion is needed, but only handful of fields.
Once I implement ( which may not be that soon ) the `*` field, then we can have rather flexible query language with optional expansion. Say
```
?fields=*,posts{*,user{*}}
```
Where `posts` and `user` are references.

**Comment by Dragomir-Ivanov** [comment]
In the mean time, I will try to add another parameter to `schema.Reference`, `Array: true` as a POC, and see where it goes.
BTW, Happy New Year @smyrman ! Wish you much success in the 2019 year.

**Comment by Dragomir-Ivanov** [comment]
I think I nailed #222 and #223, but it conflicts with yet not merged #215.
